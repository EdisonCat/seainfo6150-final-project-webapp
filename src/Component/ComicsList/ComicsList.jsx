import React, {useState, useEffect} from "react";
import styles from "./ComicsList.module.css";
import ComicsCard from "../ComicsCard/ComicsCard";
export default function ComicsList() {
    const [items, setItems] = useState([]);
    const [page, setPage] = useState(0);
    const [loading, setLoading] = useState(true);
    useEffect(() => {
        setLoading(true);
        const fetchData = async () => {
            const url = `${process.env.REACT_APP_MARVEL_COMICS_PATH}${process.env.REACT_APP_MARVEL_API_HASH}&offset=${page}`;
            const res = await fetch(url, {
                method: "GET", 
                headers: {"Content-Type" : "application/json"}
            });
            const json = await res.json();
            setItems(json.data.results);
            setLoading(false);
        }
        fetchData();
    }, [page]);
    return (
        <div className={styles.container}>
            {
                loading ? "Loading..." : 
            <div className={styles.list}>
                {items.map((item, idx) => <ComicsCard key={idx} item={item}/>)}
            </div>
            }
			<div className={styles.cta}>
				{page === 0 ? null : (
					<button onClick={() => setPage(page - 20)}>Previous</button>
				)}
				<button onClick={() => setPage(page + 20)}>Next</button>
			</div>
        </div>
    );
}