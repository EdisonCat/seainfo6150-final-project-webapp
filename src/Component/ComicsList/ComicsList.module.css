.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    min-height: 900px;
    background: #333;
    border-top: 1px gray solid;
    color: white;
}
.cta {
    display: flex;
    align-items: center;
    justify-content: center;
}
.cta button {
    background: white;
    width: 100px;
    transition: 0.5s;
}
.cta button:hover {
    background: red;
    width: 120px;
    color: white;
}