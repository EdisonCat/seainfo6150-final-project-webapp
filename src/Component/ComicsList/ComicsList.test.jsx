import React from "react";
import ComicsList from "./ComicsList";
it("rendered ComicsList correctly", () => {
	// eslint-disable-next-line no-undef
    const { container } = render(<ComicsList />);
    expect(container).toMatchSnapshot();
});