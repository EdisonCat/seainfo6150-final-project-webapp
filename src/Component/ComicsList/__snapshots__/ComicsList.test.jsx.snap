// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered ComicsList correctly 1`] = `
<div>
  <div
    class="container"
  >
    Loading...
    <div
      class="cta"
    >
      <button>
        Next
      </button>
    </div>
  </div>
</div>
`;
